x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/EsqlSpecTestCase.java

```diff
@@ -182,6 +186,16 @@ protected static void checkCapabilities(RestClient client, TestFeatureService te
                 throw e;
             }
         }
+
+        var features = Stream.concat(
+            new EsqlFeatures().getFeatures().stream(),
+            new EsqlFeatures().getHistoricalFeatures().keySet().stream()
+        ).map(NodeFeature::id).collect(Collectors.toSet());
+        assumeTrue(
+            "All capabilities must be features for this cluster to work with them",
```

Comment:
this message needs to say something like "this isn't an ESQL capability or something"

Reply:
This seems to imply that we cannot add capabilities that are not actually features. At least for as long as this fall-back will be used, which seems to be for BWC tests in mixed cluster against 8.13.x and 8.14.x (based on the failures I'm seeing in CI).

Reply:
Just that we can't fall back to capabilities that aren't ESQL features.

Reply:
OK. My mistake. I realize now that the message is misleading, since the actual behaviour will just disable the test, which is the desired behaviour.